cmd/gateway/azure/gateway-azure.go

```diff
@@ -46,12 +46,14 @@ import (
 )
 
 const (
-	globalAzureAPIVersion      = "2016-05-31"
-	azureBlockSize             = 100 * humanize.MiByte
-	azureS3MinPartSize         = 5 * humanize.MiByte
-	metadataObjectNameTemplate = minio.GatewayMinioSysTmp + "multipart/v1/%s.%x/azure.json"
-	azureBackend               = "azure"
-	azureMarkerPrefix          = "{minio}"
+	globalAzureAPIVersion          = "2016-05-31"
+	azureBlockSize                 = 100 * humanize.MiByte
+	azureS3MinPartSize             = 5 * humanize.MiByte
+	metadataObjectNameTemplate     = minio.GatewayMinioSysTmp + "multipart/v1/%s.%x/azure.json"
+	metadataObjectPartNameTemplate = minio.GatewayMinioSysTmp + "multipart/v1/%s.%x.%x/azure.json"
```

Comment:
We were encoding all the information into blockIDs itself so that we don't have to store an external metadata file like this. But now if we are having a part-meta file for each part then we don't have to encode anything in blockIDs and instead encode all the info in this meta file itself. This way blockIDs are just UUIDs and the part-meta file will look like this:

metadataObjectPartNameTemplate = minio.GatewayMinioSysTmp + "multipart/v1/%s.%x.%x/part-%d.json"
```
{
    etag: "d9cca721a735dac4efe709e0f3518373",
    blockIDs: ["4ef280b9-5752-49f8-ab6c-bc464c7dde0e", "0beb2b5a-633e-410a-a43e-6725bc366ef4", ....]
}
```

Also the following naming conventions will make it harder for future maintainers:
```
	metadata["x-amz-meta-etag"] = etag
	metadata["x-amz-meta-md5sum"] = md5Hex
```

i.e here x-amz-meta-etag is the unique string we use for azure and x-amz-meta-md5sum is the etag that is returned to s3-client. We can clean it up if we make the above change. Let's discuss with @abperiasamy

Reply:
@krishnasrinivas, refactored to use the per part metadata file as per discussion for multipart operation.